src/models/logs.js

```diff
@@ -0,0 +1,431 @@
+    if (!this.descrToId.hasOwnProperty(descr)) {
+      return;
+    }
+    const id = this.descrToId[descr];
+
+    let table = METRICS_OTHER;
+    let value = rawValue;
+
+    switch (typeof rawValue) {
+      case 'boolean':
+        table = METRICS_BOOLEAN;
+        break;
+      case 'number':
+        table = METRICS_NUMBER;
+        break;
+      default:
+        value = JSON.stringify(rawValue);
+        break;
+    }
+
+    await this.run(
+      `INSERT INTO ${table} (id, date, value) VALUES (?, ?, ?)`,
+      [id, date, value]
+    );
+  }
+
+  /**
+   * Remove a metric with all its associated data
+   * @param {Object} rawDescr
+   */
+  async unregisterMetric(rawDescr) {
+    const descr = JSON.stringify(rawDescr);
+    const id = this.descrToId[descr];
+    await Promise.all([
+      'metricIds',
+      METRICS_NUMBER,
+      METRICS_BOOLEAN,
+      METRICS_OTHER,
+    ].map((table) => {
+      return this.run(`DELETE FROM ${table} WHERE id = ?`,
+                      [id]);
+    }));
+    delete this.descrToId[descr];
+    delete this.idToDescr[id];
+  }
+
+  onPropertyChanged(property) {
+    const thingId = property.device.id;
+    const descr = this.propertyDescr(thingId, property.name);
+    this.insertMetric(descr, property.value, new Date());
+  }
+
+  onEvent() {
+  }
+
+  onAction() {
+  }
+
+  buildQuery(table, id, start, end, limit) {
+    const conditions = [];
+    const params = [];
+    if (typeof id === 'number') {
+      conditions.push('id = ?');
+      params.push(id);
+    }
+    if (start || start === 0) {
+      conditions.push('date > ?');
+      params.push(start);
+    }
+    if (end) {
+      conditions.push('date < ?');
+      params.push(end);
+    }
+
+    let query = `SELECT id, value, date FROM ${table}`;
+    if (conditions.length > 0) {
+      query += ' WHERE ';
+      query += conditions.join(' AND ');
+    }
+    if (limit) {
+      query += ` LIMIT ${limit}`;
+    }
+    return {
+      query,
+      params,
+    };
+  }
+
+  async loadMetrics(out, table, transformer, id, start, end) {
+    const {query, params} = this.buildQuery(table, id, start, end);
+    const rows = await this.all(query, params);
+
+    for (const row of rows) {
+      const descr = this.idToDescr[row.id];
+      if (!out.hasOwnProperty(descr.thing)) {
+        out[descr.thing] = {};
+      }
+      if (!out[descr.thing].hasOwnProperty(descr.property)) {
+        out[descr.thing][descr.property] = [];
+      }
+      const value = transformer ? transformer(row.value) : row.value;
+      out[descr.thing][descr.property].push({
+        value: value,
+        date: row.date,
+      });
+    }
+  }
+
+  async getAll(start, end) {
+    const out = {};
+    await this.loadMetrics(out, METRICS_NUMBER, null, null, start, end);
+    await this.loadMetrics(out, METRICS_BOOLEAN, (value) => !!value, null,
+                           start, end);
+    await this.loadMetrics(out, METRICS_OTHER, (value) => JSON.parse(value),
```

Comment:
value is a JSON-formatted string so that metricsOther can eventually store the payload object from actions and events in addition to string values like color and log messages. JSON.stringify is used over `${value}` because the interpolated string uses the default toString which will be the always helpful "[object Object]"